### DISCUSSION TITLE
Custom QR code for WalletConnect

### DISCUSSION THREAD

**Discussion by matheuseabra** [wevm/wagmi#920] [Q&A]
Hi wagmi team,

I was wondering if there's an option to not use the standard WalletConnect QR code modal that is used, I know there's a `qrcode: boolean;` option on:

```
const connector = new WalletConnectConnector({
  options: {
    qrcode: false,
  },
})
```

When I pass this option it doesn't anything else, how could I show a custom QR code?

**Comment by tmm** [comment]
One approach is listening for the [`message` event with `type: connecting`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/connectors/walletConnect.ts#L62), then getting the WalletConnect session URI (e.g. `(await connector.getProvider()).connector.uri`) and displaying that in QR code. [RainbowKit](https://github.com/rainbow-me/rainbowkit) is open source and has an example of this if you poke around the code.
